code/game/objects/items/devices/laserpointer.dm

```diff
@@ -68,103 +132,133 @@
+	for(var/mob/living/carbon/human/target_felinid in view(1, targloc))
+		if(!isfelinid(target_felinid) || target_felinid.stat == DEAD || target_felinid.is_blind() || target_felinid.incapacitated())
 			continue
-		if(user.body_position == STANDING_UP)
-			H.setDir(get_dir(H,targloc)) // kitty always looks at the light
+		if(target_felinid.body_position == STANDING_UP)
+			target_felinid.setDir(get_dir(target_felinid, targloc)) // kitty always looks at the light
 			if(prob(effectchance * diode.rating))
-				H.visible_message(span_warning("[H] makes a grab for the light!"),span_userdanger("LIGHT!"))
-				H.Move(targloc)
-				log_combat(user, H, "moved with a laser pointer",src)
+				target_felinid.visible_message(span_warning("[target_felinid] makes a grab for the light!"), span_userdanger("LIGHT!"))
+				target_felinid.Move(targloc)
+				log_combat(user, target_felinid, "moved with a laser pointer", src)
 			else
-				H.visible_message(span_notice("[H] looks briefly distracted by the light."), span_warning("You're briefly tempted by the shiny light..."))
+				target_felinid.visible_message(span_notice("[target_felinid] looks briefly distracted by the light."), span_warning("You're briefly tempted by the shiny light..."))
 		else
-			H.visible_message(span_notice("[H] stares at the light."), span_warning("You stare at the light..."))
+			target_felinid.visible_message(span_notice("[target_felinid] stares at the light."), span_warning("You stare at the light..."))
 
-	//cats!
-	for(var/mob/living/simple_animal/pet/cat/C in view(1,targloc))
+	//cats! - chance for any cat near the target to pounce at the light, stepping to the target
+	for(var/mob/living/simple_animal/pet/cat/target_kitty in view(1, targloc))
+		if(target_kitty.stat == DEAD)
+			continue
 		if(prob(effectchance * diode.rating))
-			if(C.resting)
-				C.set_resting(FALSE, instant = TRUE)
-			C.visible_message(span_notice("[C] pounces on the light!"),span_warning("LIGHT!"))
-			C.Move(targloc)
-			C.Immobilize(1 SECONDS)
+			if(target_kitty.resting)
+				target_kitty.set_resting(FALSE, instant = TRUE)
+			target_kitty.visible_message(span_notice("[target_kitty] pounces on the light!"), span_warning("LIGHT!"))
+			target_kitty.Move(targloc)
+			target_kitty.Immobilize(1 SECONDS)
 		else
-			C.visible_message(span_notice("[C] looks uninterested in your games."),span_warning("You spot [user] shining [src] at you. How insulting!"))
+			target_kitty.visible_message(span_notice("[target_kitty] looks uninterested in your games."), span_warning("You spot [user] shining [src] at you. How insulting!"))
 
-	//laser pointer image
+	//The pointer is shining, change its sprite to show
 	icon_state = "pointer_[pointer_icon_state]"
+
+	//setup pointer blip
 	var/mutable_appearance/laser = mutable_appearance('icons/obj/weapons/guns/projectiles.dmi', pointer_icon_state, 10)
```

Comment:
i also remember trying to figure out why the pointer image was not layering over anything except tiles and atmos devices, it's still a problem even with the `mutable_appearance` thing